asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py

```diff
@@ -74,77 +84,40 @@ def provenance(self) -> dict[str, str]:
     def cache(
         prepped_complexes: list[PreppedComplex],
         dir: Union[str, Path],
-        type=CacheType.DesignUnit,
     ) -> None:
         """
-        Cache a set of design units for use later.
+        Cache the list of PreppedComplex in its own folder. Each is saved as a JSON, oedu PDB and ligand SDF for vis.
         """
         dir = Path(dir)
         if not dir.exists():
             dir.mkdir(parents=True)
 
         for pc in prepped_complexes:
-            if type == CacheType.DesignUnit:
-                du_name = pc.target.target_name + ".oedu"
-                du_path = dir / du_name
-                if not du_path.exists():
-                    pc.target.to_oedu_file(du_path)
-            elif type == CacheType.JSON:
-                json_name = pc.target.target_name + ".json"
-                json_path = dir / json_name
-                if not json_path.exists():
-                    pc.to_json_file(json_path)
+            # create a folder for the complex data
+            complex_folder = dir.joinpath(f"{pc.target.target_name}-{pc.hash()}")
```

Comment:
could we make this logic a method of PreppedComplex or Complex? We will need to re-use it in the docking workflow probably. 

Something like `pc.to_unique_folder()` naming up to you.